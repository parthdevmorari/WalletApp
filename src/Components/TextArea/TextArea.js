import React from 'react';
import { TextInput, View ,Text} from 'react-native';
import { theme } from '../../Constants/Theme';
import styles from './TextAreaStyle';


const TextArea = ({ label,labelStyle, value,inputContainerStyle, mainContainerStyle,textFieldStyle,multiline, placeholder,onChangeText,keyboardType, editable,onFocus,onRef,secureTextEntry,error,onBlur,disable,length, ...props }) => (
	<View>
					<View style={[ styles.indexContainer,mainContainerStyle]}>
						<View style={styles.labelCol}>
							<Text numberOfLines={1} style={[styles.label,labelStyle]}>{label}</Text>
						</View>
						<View style={[styles.InputCol,inputContainerStyle]}>
							<TextInput
								maxLength={length} 
								keyboardType={keyboardType?keyboardType:'default'} 
								secureTextEntry={secureTextEntry ?secureTextEntry : false}
								style={[ styles.indexInput, textFieldStyle,error?{color:'red'}:{}]}
								underlineColorAndroid={'transparent'}
								onChangeText={(text) => { onChangeText(text); 	}}
								value={value}
								editable={editable}
								placeholder={placeholder}
								placeholderTextColor={error ? 'red':theme.PLACEHOLDER_1}
								ref={onRef != undefined?onRef:null}
								onFocus={()=>{ if(onFocus !== undefined) { onFocus();} }}
								onBlur={()=>{ if(onBlur !== undefined) { onBlur(); } }} 
								multiline={true}
									{...props} />
							
						</View>
					</View>
					
				</View>
);

export default TextArea;