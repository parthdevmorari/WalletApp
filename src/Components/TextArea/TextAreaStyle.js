import { StyleSheet, Dimensions, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../Constants/Color';
import { theme } from "../../Constants/Theme";

const width = Dimensions.get('window').width;
const styles = StyleSheet.create({
	indexContainer:{
		flex: 1,
		marginBottom:wp(5),
    },
    labelCol:{
        // flex: 0.6,
        marginBottom: 5,
	},
	label:{
        color: theme.LABLE_COLOR,
		fontWeight: '700',
		fontSize:wp(4),
    },
    InputCol:{
        // flex: 2.4,
        backgroundColor: '#fff',
		borderColor: '#fff',
		width: '100%',
		borderRadius: 0,
		marginBottom: 0,
		borderWidth:1,
		borderColor:theme.PRIMARY_BORDER_COLOR,
    },
    
    indexInput:{
        textAlignVertical: 'top',
		fontSize: wp(4),
		color:theme.PRIMARY_TEXT_COLOR_BLACK,
		borderStyle: 'solid',
        minHeight: 100,
        maxHeight: 'auto',
		paddingLeft: 10,
		borderColor:theme.PRIMARY_BORDER_COLOR,
    },
    

});
export default styles;

