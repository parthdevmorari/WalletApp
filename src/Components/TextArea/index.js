import component from './TextArea';
export default component;