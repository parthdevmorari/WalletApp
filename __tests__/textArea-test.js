import 'react-native';
import React from 'react';
import TextArea from '../src/Components/TextArea/TextArea'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('TextArea Component', () => {
  test('TextArea Component', () => {
    const wrapper = shallow(<TextArea />);
    expect(wrapper.exists()).toBe(true);
  });
  it('TextArea  toMatchSnapshot', () => {
    const wrapper = shallow(<TextArea />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});