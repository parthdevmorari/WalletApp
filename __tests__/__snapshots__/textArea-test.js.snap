// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component TextArea  toMatchSnapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "marginBottom": 37.5,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 30,
              "fontWeight": "700",
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 1,
            "borderColor": "#cbcbcb",
            "borderRadius": 0,
            "marginBottom": 0,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <TextInput
        keyboardType="default"
        multiline={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#B2B2B2"
        secureTextEntry={false}
        style={
          Array [
            Object {
              "borderColor": "#cbcbcb",
              "borderStyle": "solid",
              "color": "#000000",
              "fontSize": 30,
              "maxHeight": "auto",
              "minHeight": 100,
              "textAlignVertical": "top",
            },
            undefined,
            Object {},
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
